### PR TITLE
Add initial implementation of OrderShippingProvider and useOrderShipping

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### What problem is this solving?

We need to centralize all the Shipping related info to share among UI components. This PR does it.

#### How should this be manually tested?

[Workspace](https://ordershipping--vtexgame1.myvtex.com/cart)

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] [Linked this PR to a Clubhouse story (if applicable).](https://app.clubhouse.io/vtex/story/18320/implementar-consumer-do-order-shipping-no-shipping-calculator)
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️| New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/OrderShipping.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+interface OrderShippingProps {
+  children: ReactNode
+}
+
+const OrderShippingContext = createContext<Context | undefined>(undefined)
+
+export const OrderShippingProvider = ({ children }: OrderShippingProps) => {
+  const { orderForm } = useOrderForm()
```
Comment (marcoskwkm) [inline react/OrderShipping.tsx]: nit:
```ts
const { orderForm: { shipping: { countries } } } = useOrderForm()
```

**Review by marcoskwkm — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,7 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
-## Changed
-- Removed `injectIntl` from example
+## Added
+
+- Initial implementation of `OrderShippingProvider` and `useOrderShipping`.
 
 - **Component** Create the VTEX Store Component _IO Base App_
```
Comment (marcoskwkm) [inline CHANGELOG.md]: Remove this.

Comment (marcoskwkm) [inline CHANGELOG.md]: Remove this.

**Review by marcoskwkm — CHANGES_REQUESTED** [review]

**Comment by marcoskwkm** [comment]
Also, consider adapting this repo to conform with the [IO Docs Builder](https://github.com/vtex-apps/io-documentation).

**Review by marcoskwkm — APPROVED** [review]

**Review by darlenedms — APPROVED** [review]

**Pull request merged**